src/audio/mux/mux.c

```diff
@@ -610,48 +610,116 @@ static int mux_copy(struct comp_dev *dev)
 
 static int mux_reset(struct comp_dev *dev)
 {
+	int dir = dev->pipeline->source_comp->direction;
+	struct list_item *blist;
+	struct comp_buffer *source;
+
 	comp_info(dev, "mux_reset()");
 
-	return comp_set_state(dev, COMP_TRIGGER_RESET);
+	if (dir == SOF_IPC_STREAM_PLAYBACK) {
+		list_for_item(blist, &dev->bsource_list) {
+			source = container_of(blist, struct comp_buffer,
+					      sink_list);
+			/* only mux the sources with the same state with mux */
+			if (source->source->state > COMP_STATE_READY)
+				/* should not reset the downstream components */
+				return 1;
```

Comment:
I think we have a macro for this result ?

Reply:
@lgirdwood sorry didn't get it.. so what part and macro we're you thinking?

Reply:
Check pipeline.h, there are some macros for pipeline return macros when walked. Perhaps this should be an enum in the future though.

Reply:
hmm maybe PPL_STATUS_PATH_STOP then..